### PR TITLE
Add `mapTableToClass` method to create custom mapping between data tables and client classes

### PR DESCRIPTION
**Pull request by stanislav-grin**
**Usage:**
```js
class PlaceToEat {
  constructor(props = {}) {
    let { name, ownerId, objectId, cuisine, updated, created, location, owner } = props

    this.objectId = objectId
    this.name = name
    this.ownerId = ownerId
    this.cuisine = cuisine
    this.updated = updated
    this.created = created
    this.owner = owner
    this.location = location
  }

  // skipping getters and setters for brevity
}

class Information {
  constructor(props = {}) {
    const { streetAddress, country, objectId, updated, created, ownerId, city, phoneNumber, menu } = props

    this.objectId = objectId
    this.streetAddress = streetAddress
    this.country = country
    this.updated = updated
    this.created = created
    this.ownerId = ownerId
    this.city = city
    this.phoneNumber = phoneNumber
    this.menu = menu
  }

  // skipping getters and setters for brevity
}

Backendless.Data.mapTableToClass('Restaurant', PlaceToEat)
Backendless.Data.mapTableToClass('Location', Information)

const query = Backendless.DataQueryBuilder.create()
  .addRelated('location')

Backendless.Data.of('Restaurant').find(query)
```

**Results:**
```
[ PlaceToEat {
    objectId: '55070490-CE2D-6C7E-FF97-99CF012E5700',
    name: 'Beef & Wine',
    ownerId: '0D9F870F-666E-FDE1-FF84-7731F3498200',
    cuisine: undefined,
    updated: 1569919487240,
    created: 1569919463225,
    owner: undefined,
    location:
     Information {
       objectId: '3086C565-DCC9-949D-FF55-CAD1E02BF200',
       streetAddress: undefined,
       country: undefined,
       updated: null,
       created: 1569919527652,
       ownerId: null,
       city: undefined,
       phoneNumber: undefined,
       menu: undefined,
       ___class: 'Location' },
    ___class: 'Restaurant' } ]
```

### CODE REVIEW

File: src/data/store/index.js [hunk 1]
```diff
@@ -15,14 +15,16 @@ const namespaceLabel = 'Backendless.Data.of(<ClassName>)'
 
 class DataStore {
 
-  constructor(model) {
+  constructor(model, classToTableMap) {
```
Comment (Valodya) [inline src/data/store/index.js]: what do you thing about renaming `classToTableMap` -> `classesMapping` ?

File: src/data/index.js [hunk 2]
```diff
@@ -29,8 +31,15 @@ const Data = {
   @deprecated('Backendless.Data', 'Backendless.Data.of(<ClassName>).save')
   saveSync(className, obj, asyncHandler){
     return this.of(className).saveSync(obj, asyncHandler)
-  }
+  },
 
+  mapTableToClass(tableName, clientClass) {
+    if (!clientClass) {
```
[the remark this answers is not in the document]
Comment (stanislav-grin) [reply src/data/index.js]: Yes, agree

File: src/data/store/index.js (hunk 1 above)
Comment (stanislav-grin) [reply src/data/store/index.js]: Why do you think that would be better? IMO with the current version of the name the relativity of relations becomes clear.

Comment (Valodya) [reply src/data/store/index.js]: ok, leave this name, looks like you are right

**Comment by Valodya** [comment]
have no comments any more, merge it down

**Pull request merged**